File: src/main/java/co/herod/adbwrapper/model/AdbUiHierarchy.java
package co.herod.adbwrapper.model;

public class AdbUiHierarchy {

    private final String xmlString;
    private final AdbDevice adbDevice;

    public AdbUiHierarchy(final String xmlString, final AdbDevice adbDevice) {

        this.xmlString = xmlString;
        this.adbDevice = adbDevice;
    }

    public String getXmlString() {
        return xmlString;
    }

    public AdbDevice getAdbDevice() {
        return adbDevice;
    }

    @Override
    public String toString() {
        return "AdbUiHierarchy{" +
                "xmlString='" + xmlString + '\'' +
                ", adbDevice=" + adbDevice +
                '}';
    }
}

File: adb-kotlin/src/main/kotlin/co/herod/adbwrapper/model/AdbUiHierarchy.kt
@file:Suppress("MemberVisibilityCanPrivate")

package co.herod.adbwrapper.model

import java.util.*

class AdbUiHierarchy(
        val xmlString: String,
        val adbDevice: AdbDevice?,
        val dumpDate: Date = Date()
) {
    override fun toString(): String {
        return "AdbUiHierarchy(xmlString='$xmlString', adbDevice=$adbDevice, dumpDate=$dumpDate)"
    }
}
